triggers/daml/AutoApproval.daml

```diff
@@ -0,0 +1,132 @@
+  distributionRoleOffers    <- generateCommands DistributionRole.Accept   <$> queryByField @DistributionRole.Offer    @"provider" (.provider) (== party)
+  distributionRoleRequests  <- generateCommands DistributionRole.Approve  <$> queryByField @DistributionRole.Request  @"operator" (.operator) (== party)
+  issuanceServiceOffers     <- generateCommands IssuanceService.Accept    <$> queryByField @IssuanceService.Offer     @"customer" (.customer) (== party)
+  listingServiceOffers      <- generateCommands ListingService.Accept     <$> queryByField @ListingService.Offer      @"customer" (.customer) (== party)
+  regulatorServiceOffers    <- generateCommands RegulatorService.Accept   <$> queryByField @RegulatorService.Offer    @"customer" (.customer) (== party)
+  settlementServiceOffers   <- generateCommands SettlementService.Accept  <$> queryByField @SettlementService.Offer   @"provider" (.provider) (== party)
+  settlementServiceRequests <- generateCommands SettlementService.Approve <$> queryByField @SettlementService.Request @"operator" (.operator) (== party)
+  tradingRoleOffers         <- generateCommands TradingRole.Accept        <$> queryByField @TradingRole.Offer         @"provider" (.provider) (== party)
+  tradingRoleRequests       <- generateCommands TradingRole.Approve       <$> queryByField @TradingRole.Request       @"operator" (.operator) (== party)
+  matchingserviceOffers     <- generateCommands MatchingService.Accept    <$> queryByField @MatchingService.Offer     @"provider" (.provider) (== party)
+  matchingserviceRequests   <- generateCommands MatchingService.Approve   <$> queryByField @MatchingService.Request   @"operator" (.operator) (== party)
+
+  -- Exercise a choice on a contract (ie, a role contract) which isn't the triggered contract
+  custodyServicesRequests   <- queryByField @CustodyService.Request  @"provider" (.provider) (== party)
+  issuanceServiceRequests   <- queryByField @IssuanceService.Request @"provider" (.provider) (== party)
+  custodyRoles              <- queryByField @CustodyRole.Role        @"provider" (.provider) (== party)
+  let custodyRequests  = generateRoleCommands (\cid -> CustodyRole.AcceptCustodyRequest with custodyRequestCid = cid) custodyServicesRequests custodyRoles
+      issuanceRequests = generateRoleCommands (\cid -> CustodyRole.AcceptIssuanceRequest with issuanceServiceRequestCid = cid) issuanceServiceRequests custodyRoles
+
+  auctionServiceRequests    <- queryByField @AuctionService.Request @"provider" (.provider) (== party)
+  biddingServiceRequests    <- queryByField @BiddingService.Request @"provider" (.provider) (== party)
+  distributionRole          <- queryByField @DistributionRole.Role  @"provider" (.provider) (== party)
+  let auctionRequests = generateRoleCommands (\cid -> DistributionRole.ApproveAuctionServiceRequest with auctionServiceRequestCid = cid) auctionServiceRequests distributionRole
```

Comment:
```suggestion
  let auctionRequests = generateRoleCommands DistributionRole.ApproveAuctionServiceRequest auctionServiceRequests distributionRole
```

Reply:
Good catch - I'll apply this to the other relating variables too!